admin/server/github.go

```diff
@@ -88,53 +134,254 @@ func (s *Server) githubConnect(w http.ResponseWriter, r *http.Request, pathParam
 		return
 	}
 
-	// NOTE: If needed, we can add a `state` query parameter that will be passed through to githubConnectCallback.
+	query := r.URL.Query()
+	remote := query.Get("remote")
+	// Should we add any other validation for remote ?
+	if remote == "" {
```

Comment:
1. nit: can just do `r.URL.Query().Get("remote")`, no need for extra var
2. About validation, should we just make `remote` optional? In case we ever want to redirect to just let people edit permissions generally

Reply:
1. I prefer(not a hard rule) to create local vars instead of chaining functions especially when first function is not returning any pre-computed data. Preferable from a debugging point of view. 
2. Yeah makes sense. I will redirect to connect success page in this case.

Reply:
1. My two cents: when reading code, seeing a variable declaration is more cognitive overhead for me, because I look through the scope to understand it's lifecycle and how it will be mutated. If I need to debug something, I just briefly edit the code during debugging only if necessary (rarely necessary).